vendor-extra/LiberoPageBundle/src/EventListener/BuildView/ItemRefTeaserHrefListener.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Libero\LiberoPageBundle\EventListener\BuildView;
+
+use FluentDOM\DOM\Element;
+use Libero\ViewsBundle\Views\SimplifiedViewConverterListener;
+use Libero\ViewsBundle\Views\TemplateView;
+use Libero\ViewsBundle\Views\View;
+use Symfony\Component\Routing\Generator\UrlGeneratorInterface;
+use function Libero\ViewsBundle\array_has_key;
+
+final class ItemRefTeaserHrefListener
+{
+    use SimplifiedViewConverterListener;
+
+    private $urlGenerator;
+
+    public function __construct(UrlGeneratorInterface $urlGenerator)
+    {
+        $this->urlGenerator = $urlGenerator;
+    }
+
+    protected function handle(Element $object, TemplateView $view) : View
+    {
+        /** @var string $id */
+        $id = $object->getAttribute('id');
+        /** @var string $service */
+        $service = $object->getAttribute('service');
+
+        if ('' === $id || '' === $service) {
+            return $view;
+        }
+
+        return $view->withArgument(
+            'href',
+            $this->urlGenerator->generate("libero.page.content.{$service}", ['id' => $id])
```

Comment:
good abstraction of UrlGenerator as an application service that is passed in where needed